### PR TITLE
Backend projects pagination

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary
Can be deployed to prod without any frontend changes and won't have any effect

```
{
  projectsCount {
    projectsCount
    erc20ProjectsCount
    currencyProjectsCount
  }
  
  all: allErc20Projects{
    name
  }
  
  paginated: allErc20Projects(page: 1, pageSize: 1){
    name
  }
}
```

```
{
  "data": {
    "all": [
      {
        "name": "Libra"
      },
      {
        "name": "Santiment"
      }
    ],
    "paginated": [
      {
        "name": "Libra"
      }
    ],
    "projectsCount": {
      "currencyProjectsCount": 36,
      "erc20ProjectsCount": 2,
      "projectsCount": 38
    }
  }
}
```

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase_web/graphql/resolvers/project/project_resolver.ex [hunk 1]
```diff
@@ -11,67 +11,71 @@ defmodule SanbaseWeb.Graphql.Resolvers.ProjectResolver do
...
+    page_size = Map.get(args, :page_size, nil)
+
     erc20_projects =
-      Project.erc20_projects()
-      |> Repo.preload([
-        :latest_coinmarketcap_data,
-        icos: [ico_currencies: [:currency]]
-      ])
-      |> Enum.dedup()
+      if page == nil || page_size == nil do
```
Comment (tspenov) [inline lib/sanbase_web/graphql/resolvers/project/project_resolver.ex]: please also check for negative and anything that will not make sense for `limit` and `offset`
 here:
``` limit: ^page_size,
    offset: ^((page - 1) * page_size),
```

File: lib/sanbase/model/project/project_list.ex [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      ]
+    )
+    |> Repo.all()
+  end
+
+  @doc ~s"""
+  Return all projects
+  """
+  def projects() do
+    # The opposite of ERC20. Classify everything except ERC20 as Currency.
```
Comment (tspenov) [inline lib/sanbase/model/project/project_list.ex]: leftover comment

File: lib/sanbase/model/project/project_list.ex [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      ]
+    )
+    |> Repo.all()
+  end
+
+  @doc ~s"""
+  Return all currency projects
+  """
+  def currency_projects() do
+    # The opposite of ERC20. Classify everything except ERC20 as Currency.
```
Comment (tspenov) [inline lib/sanbase/model/project/project_list.ex]: you can move this to the docs string

File: lib/sanbase/model/project/project_list.ex [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      where:
+        not is_nil(p.coinmarketcap_id) and (is_nil(p.main_contract_address) or infr.code != "ETH")
+    )
+  end
+
+  @doc ~s"""
+  Returns `page_size` number of currency projects from the `page` pages
+  """
+  def currency_projects_page(page, page_size) do
+    # The opposite of ERC20. Classify everything except ERC20 as Currency.
```
Comment (tspenov) [inline lib/sanbase/model/project/project_list.ex]: leftover

File: lib/sanbase/model/project/project_list.ex (hunk 2 above)
Comment (IvanIvanoff) [reply lib/sanbase/model/project/project_list.ex]: It is an old comment that was added intentionally

**Pull request merged**